tests/foreman/api/test_contentviewversion.py

```diff
@@ -129,3 +133,64 @@ def test_delete_version_negative(self):
             content_view.version[0].delete()
         # Make sure that content view version is still present
         self.assertEqual(len(content_view.read().version), 1)
+
+    def test_incremental_update_puppet(self):
+        """@Test: Incrementally update a CVV with a puppet module.
+
+        @Assert: The the incremental update succeeds with no errors, and the
+        content view is given an additional version.
+
+        @Feature: ContentViewVersion
+
+        """
+        product = entities.Product().create()
+        repository = entities.Repository(
+            content_type='puppet',
+            product=product,
+        ).create()
+        with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
+            repository.upload_content(files={'content': handle})
+
+        # Create a content view version.
+        content_view = entities.ContentView(
+            organization=product.organization,
+        ).create()
+        content_view.publish()
+        content_view = content_view.read()
+
+        # Find the puppet module we uploaded in to `repository`.
+        puppet_module = entities.PuppetModule(
+            id=content_view.available_puppet_modules()['results'][0]['id']
+        )
+
+        # Incrementally update the CCV with the puppet module.
+        payload = {
+            'content_view_version_environments': [{
+                'content_view_version_id': content_view.version[0].id,
+                'environment_ids': [
+                    environment.id
+                    for environment
+                    in content_view.version[0].read().environment
+                ],
+            }],
+            'description': 'I have been incrementally updated.',
```

Comment:
Whoops. I don't think this is necessary. I'll drop this and re-post some test results.